### DISCUSSION TITLE
How to conect saleor to AWS s3? Problem with image uploading in the dashboard.

### DISCUSSION THREAD

**Discussion by demodes** [saleor/saleor#10038] [Q&A]
Hi. I deployed Saleor (core and dashboard) on Heroku. I used version 3.4.1. After deployment, the storefront and dashboard show no images. I tried to connect to AWS s3, but the images still do not show. 

In Heroku I set these config vars:
AWS_ACCESS_KEY_ID
AWS_MEDIA_BUCKET_NAME
AWS_QUERYSTRING_AUTH : true
AWS_SECRET_ACCESS_KEY
AWS_STORAGE_BUCKET_NAME

In AWS s3 I set the bucket to be public:
In Permission tab:
Access
 Public
Block all public access
 Off
I do not set 'Bucket policy' field

And I also set the CORS according to Saleor docs:
CORS set in AWS:
[
  {
    "AllowedHeaders": [
      "*"
    ],
    "AllowedMethods": [
      "GET",
      "HEAD"
    ],
    "AllowedOrigins": [
      "*"
    ],
    "ExposeHeaders": [],
    "MaxAgeSeconds": 3000
  }
]

After I upload the image in the Saleor dashboard no error is shown in the browser console. Image is visible in productDetails (in dashboard) in the media section (just under the upload image button). Here it will be visible for some time (let's say 1 hour maybe) and then it will disappear and instead there will be a placeholder. After I upload the image in the Saleor dashboard, I do not see the image in the AWS bucket dashboard.
I wouldn't be surprised If I am making something wrong, but I can not find out what it is.
![saleor-dashboard-images](https://user-images.githubusercontent.com/23063636/174867512-d1aa8f39-9c20-4695-8178-714fe44672be.png)
![saleor-heroku-config-vars](https://user-images.githubusercontent.com/23063636/174867535-6abc361b-5fa5-4ac2-a1c5-a99f6235a808.png)
![saleor-aws-permission-1](https://user-images.githubusercontent.com/23063636/174867557-6d08a9bd-0fd5-4bdb-b5be-a662335f78c1.png)
![saleor-aws-permission-2](https://user-images.githubusercontent.com/23063636/174867569-e075f9e0-3c0a-41c3-8b0d-73d9ed208ae7.png)

**Comment by timuric** [comment]
@denisdemolli do images disappear from the AWS as well?

**Reply by demodes** [reply]
The images never shown in AWS. 
After I upload the image in a dashboard, the image is visible in dashboard for some short time (maybe 30-60 min). After that time the image dissapier. The image never show in storefront. But other changes in dashboard (creating new product, etc.) are working and are visible in storefront.

Can it be that images are saved to Heroku temporary storage  and because of that they keep disappearing and also not show in AWS. In that case it might be some mistake in my aws integration settings, but I can not see it.

**Comment by karolkielecki** [comment]
Hi @denisdemolli, Perhaps you can find the answer under the following Spectrum thread. https://spectrum.chat/saleor/general/images-upload-successfully-to-aws-s3-bucket-but-do-not-display~2d444e74-f294-42d5-ae8b-c8c24d232bc0

**Reply by demodes** [reply]
In the thread is mentioned some licensing issue in the S3 bucket. But the author of the comment didn't go deeper, so I don't really know what this is about.

**Reply by marorvel31** [reply]
Hi, how you achieve this finally?
